runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -667,7 +718,7 @@ protected void optimisticRollbackUnsafe() {
      * @param entry smr entry
      */
     public void applyUpdateToStreamUnsafe(SMREntry entry, long globalAddress) {
-        applyUpdateUnsafe(entry);
+        applyUpdateUnsafe(entry, globalAddress);
```

Comment:
SMREntry has `getGlobalAddress`

Reply:
Ok.

Reply:
Turns out we cannot rely on the `getGlobalAddress()`, since in some cases, special timestamp is being passed:

```
       long syncTo = (timestamp == Address.OPTIMISTIC) ? Address.MAX : timestamp;
        stream.streamUpTo(syncTo)
                .forEachOrdered(entry -> {
                    try {
                        Object res = applyUpdateUnsafe(entry, timestamp);
```